### PR TITLE
fix functional benchmarker and always test

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Fixes the functional-benchmarker to close the client as desired
* Sets up Makefile to test when we do functional tests to ensure it exits correctly

@igor-karpukhin @vimalk78 @alanconway

### CODE REVIEW

**Review by igor-karpukhin — APPROVED** [review]
/lgtm

File: test/functional/framework.go [hunk 1]
```diff
@@ -115,7 +115,7 @@ func NewCRIOLogMessage(timestamp, message string, partial bool) string {
 }
 
 func (f *FluentdFunctionalFramework) Cleanup() {
-	f.Test.Close()
+	f.closeClient()
```
Comment (vimalk78) [inline test/functional/framework.go:118]: nit: `nil` check before invoking function

**Comment by jcantrill** [comment]
@vimalk78 @alanconway @igor-karpukhin  please review

**Review by alanconway — APPROVED** [review]
/approve
/lgtm

**Pull request merged**